lib/racer-client.coffee

```diff
@@ -11,50 +11,56 @@ class RacerClient
   project_path: null
   candidates: null
 
-  check_completion: (editor, row, col, cb) ->
-    if !@process_env_vars()
-      console.error("Your racer package is not properly configured.")
-      cb null
-      return
+  check_generator = (racer_action) ->
+    (editor, row, col, cb) ->
+      if !@process_env_vars()
+        console.error("Your racer package is not properly configured.")
+        cb null
+        return
 
-    temp_folder_path = path.dirname(editor.getPath())
-    # temp_folder_path will be '.' for unsaved files
-    if temp_folder_path == "."
-      temp_folder_path = @project_path
+      temp_folder_path = path.dirname(editor.getPath())
+      original_file_name = path.basename(editor.getPath())
+      # temp_folder_path will be '.' for unsaved files
+      if temp_folder_path == "."
+        temp_folder_path = @project_path
 
-    tempOptions =
-      prefix: "._racertmp"
-      dir: temp_folder_path
+      tempOptions =
+        prefix: original_file_name + ".racertmp_"
```

Comment:
Btw, I hope the trick will still work, or we'll see flickers problems again...
Question is does Atom considers `main.rs._racertmpXXX` to have a `._*` extension.